### ISSUE TITLE
pip install --upgrade diffusers[torch] fails with error: subprocess-exited-with-error 

### ISSUE THREAD

**Issue by shivam-maurya-git** [huggingface/diffusers#6050]
Specifications :
Windows 10
Python 3
See Image for details
![image](https://github.com/huggingface/diffusers/assets/74864959/03e7e473-fe42-4c06-99cb-7baa8fa85c04)

**Comment by shivam-maurya-git** [comment]
@sayakpaul , Thanks for noticing. 
Python Version : 3.12.0
Running code in VS Code
Trace of Error
```
Collecting diffusers[torch]
  Using cached diffusers-0.24.0-py3-none-any.whl.metadata (18 kB)
Requirement already satisfied: Pillow in c:\users\apu\appdata\local\programs\python\python312\lib\site-packages (from diffusers[torch]) (10.1.0)
Collecting filelock (from diffusers[torch])
  Using cached filelock-3.13.1-py3-none-any.whl.metadata (2.8 kB)
Collecting huggingface-hub>=0.19.4 (from diffusers[torch])
  Using cached huggingface_hub-0.19.4-py3-none-any.whl.metadata (14 kB)
Collecting importlib-metadata (from diffusers[torch])
  Using cached importlib_metadata-7.0.0-py3-none-any.whl.metadata (4.9 kB)
Requirement already satisfied: numpy in c:\users\apu\appdata\local\programs\python\python312\lib\site-packages (from diffusers[torch]) (1.26.2)
Collecting regex!=2019.12.17 (from diffusers[torch])
  Using cached regex-2023.10.3-cp312-cp312-win_amd64.whl.metadata (41 kB)
Requirement already satisfied: requests in c:\users\apu\appdata\local\programs\python\python312\lib\site-packages (from diffusers[torch]) (2.31.0)
Collecting safetensors>=0.3.1 (from diffusers[torch])
  Using cached safetensors-0.4.1.tar.gz (62 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [6 lines of output]
     
      Cargo, the Rust package manager, is not installed or is not on PATH.
      This package requires Rust and Cargo to compile extensions. Install it through
      the system's package manager or via https://rustup.rs/
     
      Checking for Rust toolchain....
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

**Comment by sayakpaul** [comment]
@DN6 do we know anything about it?

**Comment by standardAI** [comment]
@shivam-maurya-git Could you install what it says and then try again?
Also, would you prefer a lower version of Python such as 3.10? I couldn't install `diffusers` with Python 3.12, it says dependency conflicts.

**Comment by shivam-maurya-git** [comment]
@standardAI I am getting same error.

**Comment by asomoza** [comment]
I think like it was discussed in #5712, it maybe necessary to put in an important place that pytorch doesn't support python 3.12 on windows and probably its going to take several months until that happens. Its not a diffusers problem but I keep seeing people with this error everywhere not just here.

TLDR: don't use python 3.12, use python 3.11

**Comment by shivam-maurya-git** [comment]
@asomoza Thanks. This error resolved using python 3.11.5
